modules/core/src/smithy4s/internals/Hinted.scala

```diff
@@ -138,6 +138,21 @@ object Hinted {
         f: (Option[H1], Option[H2]) => F[A]
     ): Hinted[F, A] =
       Hinted(Hints(), (h: Hints) => f(h.get[H1], h.get[H2]))
+    def onHintOneOf[H1: ShapeTag, H2: ShapeTag, A](
```

Comment:
Interesting. I'll allow it 😄

Reply:
personally I like to do sth like 

```scala

Hinted[F].from { hints =>
  H1(hints).map(onH1)
    ..orElse(H2(hints).map(onH2))
}
```

but there's still the `from` outside that may not be the most convenient :)

Reply:
Maybe we should generate custom extractors, and allow for : 

``` 
Hinted[F].from { 
    case H1(...) => onH1
    case H2(...) => onH2
} 
```

Reply:
I tried a top level extractor in the companion object directly but it conflicts with case class default extractors. instead, I went for something that produce this:

```scala
package smithy4s.api

import smithy4s.syntax._

case class Untagged()
object Untagged extends smithy4s.ShapeTag.Companion[Untagged] {
  val id: smithy4s.ShapeId = smithy4s.ShapeId("smithy4s.api", "untagged")

  val hints : smithy4s.Hints = smithy4s.Hints(
    id,
    smithy.api.Trait(Some("union :not([trait|smithy4s.api#discriminated])"), None, None),
  )

  val schema: smithy4s.Schema[Untagged] = constant(Untagged()).withHints(hints)
  implicit val staticSchema : schematic.Static[smithy4s.Schema[Untagged]] = schematic.Static(schema)
  
  object extractHint { def unapply(h: smithy4s.Hints): Option[Untagged] = h.get[Untagged] }
}
```

wdyt?

Reply:
see https://github.com/disneystreaming/smithy4s/pull/178/commits/778ac7296b55af18cb292688d73985cccf8f881f

Reply:
got rid of it in favor of extractors for `ShapeTag.Companion` and `NewType`